### PR TITLE
[SPARK-37825][SQL][LAUNCHER] Make spark beeline be able to handle javaOpts

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Currently, we build the beeline command with SPARK_DRIVER_MEMORY only and are not able to set extra java opts. Besides, the beeline is not a DRIVER-type thing.

In this PR, we add 

```java
# - SPARK_BEELINE_OPTS, to set config properties only for the beeline cli (e.g. "-Dx=y")
# - SPARK_BEELINE_MEMORY, Memory for beeline (e.g. 1000M, 2G) (Default: 1G)
```

### Why are the changes needed?

Make spark beeline  configurable for extra JVM settings, like GC 

### Does this PR introduce _any_ user-facing change?

yes, add 2 environment variables.

### How was this patch tested?

add a new java ut

### CODE REVIEW

File: launcher/src/test/java/org/apache/spark/launcher/SparkClassCommandBuilderSuite.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class SparkClassCommandBuilderSuite extends BaseSuite {
+
+  @Test
+  public void testBeelineBuilder() throws Exception {
+    List<String> args = Arrays.asList("myBeelineArg");
+    SparkClassCommandBuilder builder = new SparkClassCommandBuilder("org.apache.hive.beeline.BeeLine", args);
```
Comment (yaooqinn) [inline launcher/src/test/java/org/apache/spark/launcher/SparkClassCommandBuilderSuite.java:33]:
```suggestion
    SparkClassCommandBuilder builder =
      new SparkClassCommandBuilder("org.apache.hive.beeline.BeeLine", args);
```

**Comment by yaooqinn** [comment]
cc @wangyum @cloud-fan thanks

File: pom.xml [hunk 2]
```diff
@@ -2910,6 +2910,7 @@
               <SPARK_SCALA_VERSION>${scala.binary.version}</SPARK_SCALA_VERSION>
               <SPARK_TESTING>1</SPARK_TESTING>
               <JAVA_HOME>${test.java.home}</JAVA_HOME>
+              <SPARK_BEELINE_OPTS>-DmyKey=yourValue</SPARK_BEELINE_OPTS>
```
Comment (cloud-fan) [inline pom.xml:2913]: is it for test?

Comment (yaooqinn) [reply pom.xml:2913]: yes

**Review by cloud-fan — APPROVED** [review]

**Comment by yaooqinn** [comment]
let me rebase the master and test again

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**